src/Adapter/Product/Image/Repository/ProductImageMultiShopRepository.php

```diff
@@ -112,44 +112,51 @@ function (ImageId $imageId) use ($shopId): Image {
 
     /**
      * @param ProductId $productId
+     * @param ShopConstraint $shopConstraint
      *
      * @return ImageId[]
      */
     public function getImagesIds(ProductId $productId, ShopConstraint $shopConstraint): array
     {
-        $qb = $this->connection->createQueryBuilder()->select('i.id_image');
+        $qb = $this->connection->createQueryBuilder()
+            ->select('i.id_image')
+            ->from($this->dbPrefix . 'image', 'i')
+            ->andWhere('i.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->addOrderBy('i.position', 'ASC')
```

Comment:
one note - to expect consistent results I made thus always order by position (before it didn't order by position all the time)